tenant/repository_test.go

```diff
@@ -322,6 +322,93 @@ func (s *TenantServiceTestSuite) TestGetSubsetOfTenantsThatMatchesRequiredCluste
 	})
 }
 
+func (s *TenantServiceTestSuite) TestGetNumberOfAllOutdatedTenants() {
+	// given
+	configuration.Commit = "123abc"
+	testdoubles.SetTemplateVersions()
+	tf.FillDB(s.T(), s.DB, tf.AddTenants(3), tf.AddDefaultNamespaces().State(tenant.Ready).Outdated())
+	svc := tenant.NewDBService(s.DB)
```

Comment:
the name for `NewDBService` is not usual: in `auth` and `cluster` services, we have a `service` layer and a `repository` layer (this later uses GORM to run SQL queries). So `DBService` looks like a mix of both

Reply:
I know, I don't like the name from the very first moment. I don't like the name of the package either.  It's one of the things I wanted to change as part of a rewrite of this DB representation.

I could change it in the [big core PR](https://github.com/fabric8-services/fabric8-tenant/pull/673), however, there is already `Repository` introduced so I could name it `Service`. But in the combination with the package name it would result in `tenant.NewService(s.DB)` and (from my point of view) this looks weird - the name `tenant.Service` doesn't say anything about DB or repo. That's why I left the original name there for now.

Reply:
@MatousJobanek ok let's revisit this once those 2 PRs have been merged.